core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -96,7 +94,8 @@ class TransactionStateManager(brokerId: Int,
   private[transaction] val transactionMetadataCache: mutable.Map[Int, TxnMetadataCacheEntry] = mutable.Map()
 
   /** number of partitions for the transaction log topic */
-  private val transactionTopicPartitionCount = getTransactionTopicPartitionCount
+  private var retrieveTransactionTopicPartitionCount: () => Int = _
+  private var transactionTopicPartitionCount: Int = _
```

Comment:
Same question here about `@volatile`.  The value is set once at startup and is used in a denominator within `partitionFor()`